js/cd-dropdown.js

```diff
@@ -3,297 +3,302 @@
-          var wrapper = context.createElement('span');
-          wrapper.setAttribute('class', component + '__logo');
-          wrapper.appendChild(svgElem);
-          return wrapper;
-        }
-        return svgElem;
+    },
+
+    /**
+     * Create a svg icon.
+     */
+    createIcon: function (name, component, wrap) {
+      var svgElem = this.context.createElementNS('http://www.w3.org/2000/svg', 'svg');
+      var useElem = this.context.createElementNS('http://www.w3.org/2000/svg', 'use');
+      useElem.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:href', '#cd-icon--' + name);
+      svgElem.setAttribute('class', 'cd-icon cd-icon--' + name);
+      svgElem.appendChild(useElem);
+
+      if (component && wrap) {
+        var wrapper = this.context.createElement('span');
+        wrapper.setAttribute('class', component + '__logo');
+        wrapper.appendChild(svgElem);
+        return wrapper;
       }
-
-      /**
-       * Create a button to toggle a dropdown.
-       */
-      function createButton(element) {
-        var label = element.getAttribute('data-toggable');
-        var logo = element.getAttribute('data-logo');
-        var icon = element.getAttribute('data-icon');
-        var component = element.getAttribute('data-component');
-
-        // Create the button.
-        var button = context.createElement('button');
-        button.setAttribute('type', 'button');
-
-        // Pre-label logo.
-        if (logo) {
-          button.appendChild(createIcon(logo, component, true));
-        }
-
-        // Button label.
-        var labelWrapper = context.createElement('span');
-        labelWrapper.appendChild(context.createTextNode(label));
-        button.appendChild(labelWrapper);
-
-        // Post-label icon.
-        if (icon) {
-          button.appendChild(createIcon(icon));
-        }
-
-        // BEM for class selectors.
-        if (component) {
-          button.setAttribute('class', component + '__btn');
-          labelWrapper.setAttribute('class', component + '__btn-label');
-        }
-
-        // Do not collapse the dropdown when clicking outside.
-        if (element.hasAttribute('data-toggable-keep')) {
-          button.setAttribute('data-toggable-keep', '');
-        }
-
-        // Alternate label for when the button is expanded.
-        if (element.hasAttribute('data-toggable-expanded')) {
-          labelWrapper.setAttribute('data-label-switch', element.getAttribute('data-toggable-expanded'));
-        }
-
-        return button;
+      return svgElem;
+    },
+
+    /**
+     * Create a button to toggle a dropdown.
+     */
+    createButton: function (element) {
+      var label = element.getAttribute('data-toggable');
+      var logo = element.getAttribute('data-logo');
+      var icon = element.getAttribute('data-icon');
+      var component = element.getAttribute('data-component');
+
+      // Create the button.
+      var button = this.context.createElement('button');
```

Comment:
@left23 @rupl  Not sure if `context` is ever anything other than the `document` but if that's the case and it's a DOM element for example, then the `createElement` function will not exist and an error will be thrown. (Same with the createElementNS in the createIcon function etc.)

Reply:
`context` can be something other than `document`, for example when Drupal Ajax is invoked. [Here's a d.o comment explaining it](https://www.drupal.org/forum/support/module-development-and-code-questions/2013-02-19/what-is-context-for-in-drupal#comment-13074787).

However for our custom Behavior, we have reasonable certainty that it will always be `document`, and I suppose we could just simplify.